windows/src/engine/keyman/UfrmKeyman7Main.pas

```diff
@@ -844,7 +848,24 @@ function TfrmKeyman7Main.ProcessWMKeymanControl(Command, WParam: Word; LParam: D
   end;
 end;
 
-procedure TfrmKeyman7Main.DoHotkey(Target: Integer);
+procedure TfrmKeyman7Main.DoLanguageHotkey(Index: Integer);
+var
+  FKeyboard: TLangSwitchKeyboard;
+begin
+  if (Index >= 0) and (Index < kmcom.Languages.Count) then
+  begin
+    FKeyboard := FLangSwitchManager.FindKeyboard(kmcom.Languages[Index].HKL, kmcom.Languages[Index].ProfileGUID);
+    if not Assigned(FKeyboard) then Exit;
+
+    // Handle toggle hotkey
+    if (FKeyboard = FLangSwitchManager.ActiveKeyboard) and (kmcom.Options['koKeyboardHotkeysAreToggle'].Value) then
+        FKeyboard := FLangSwitchManager.Languages[0].Keyboards[0];
```

Comment:
This doesn't toggle to `Languages[Index]`?

Reply:
No, it toggles between default language/keyboard (e.g. in my case English Australia/US) and the language/keyboard associated with the hotkey.